src/handshake/server.rs

```diff
@@ -217,18 +244,24 @@ impl<'a, T: AsyncRead + AsyncWrite + Unpin> Server<'a, T> {
 /// Handshake request received from the client.
 #[derive(Debug)]
 pub struct ClientRequest<'a> {
-    ws_key: Vec<u8>,
+    ws_key: WebSocketKey,
     protocols: Vec<&'a str>,
-    path: String,
+    path: &'a str,
+    headers: RequestHeaders<'a>,
 }
 
-impl<'a> ClientRequest<'a> {
-    /// A reference to the nonce.
-    pub fn key(&self) -> &[u8] {
-        &self.ws_key
-    }
+/// Select HTTP headers sent by the client.
```

Comment:
```suggestion
/// Selected HTTP headers sent by the client.
```

Reply:
Select is a proper adjective in English, as in "select few".

Reply:
fine :)